java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S4034_java.html

```diff
@@ -1,10 +1,34 @@
-<p>The Stream API provides the <code>boolean anyMatch(Predicate&lt;? super T&gt; predicate)</code> method, which is a very convenient and efficient
-way to check whether any elements of a stream match the provided predicate.</p>
-<p>This rule raises an issue when a <code>Stream</code> chain could be replaced by this method call. Specifically:</p>
-<ul>
-  <li> <code>filter(Predicate&lt;? super T&gt; predicate).findFirst().isPresent()</code> </li>
-  <li> <code>filter(Predicate&lt;? super T&gt; predicate).findAny().isPresent()</code> </li>
-</ul>
+<p>When using the <code>Stream</code> API, call chains should be simplified as much as possible. Not only does it make the code easier to read, it
+also avoid creating unnecessary temporary objects.</p>
+<p>This rule raises an issue when one of the following substitution is possible:</p>
+<table>
+  <tbody>
+    <tr>
+      <th>Original</th>
+      <th>Preferred</th>
+    </tr>
+    <tr>
+      <td>{code}stream.filter(predicate).findFirst().isPresent(){code}</td>
+      <td>{code}stream.anyMatch(predicate){code}</td>
+    </tr>
+    <tr>
+      <td>{code}stream.filter(predicate).findAny().isPresent(){code}</td>
+      <td>{code}stream.anyMatch(predicate){code}</td>
+    </tr>
+    <tr>
+      <td>{code}!stream.anyMatch(predicate){code}</td>
+      <td>{code}stream.noneMatch(predicate){code}</td>
+    </tr>
+    <tr>
+      <td>{code}!stream.anyMatch(x -&gt; !(...)){code}</td>
+      <td>{code}stream.allMatch(...){code}</td>
+    </tr>
+    <tr>
+      <td>{code}stream.map(mapper).anyMatch(Boolean::booleanValue){code}</td>
```

Comment:
shouldn't it be "predicate" instead of "mapper" here, in order to make it explicit that you can directly use it?

Reply:
Not sure I get what you mean. `mapper` is the name of the argument in the API see https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#map-java.util.function.Function-

Reply:
Yep, but the message we are going to display is `Use predicate from "map()" directly in "anyMatch()"`